doc/GUIDE.md

```diff
@@ -0,0 +1,910 @@
+# Jextract Guide
+
+The jextract tool parses header (.h) files of native libraries, and generates Java code,
+called _bindings_, which use the [Foreign Function and Memory API](https://openjdk.org/jeps/442)
+(FFM API) under the hood, that can be used by a client to access the native library.
+
+Interacting with native C code through the FFM API works
+by loading a native library (e.g., a `.so`/`.dll`/`.dylib` file), which is essentially an
+archive of native functions and global variables. The user then has to look up the functions
+they want to call using a `SymbolLookup`, and finally _link_ the functions by using the
```

Comment:
There's some classes and methods referenced here - should we use links? (don't have strong opinion)

Reply:
I think so. This is introductory, so someone may want to read up on the FFM API before diving deeper in  jextract